src/mono/mono/profiler/browser.c

```diff
@@ -29,25 +30,107 @@ struct _MonoProfiler {
 };
 
 static MonoProfiler browser_profiler;
+static int desired_sample_interval_ms;
+static MonoCallSpec callspec;
 
-#ifdef HOST_WASM
+#ifdef HOST_BROWSER
 
-void
-mono_wasm_profiler_enter ();
+typedef struct _ProfilerStackFrame ProfilerStackFrame;
 
-void
-mono_wasm_profiler_leave (MonoMethod *method);
+struct _ProfilerStackFrame {
+	MonoMethod *method;
+	double start;
+	bool should_record;
+	ProfilerStackFrame *next;
+};
+
+static ProfilerStackFrame *profiler_stack;
```

Comment:
might want a name that more clearly communicates that this is the current stack frame instead of the topmost one

Reply:
This is actually topmost frame, I will rename it